src/catalog/settings_catalog.cpp

```diff
@@ -0,0 +1,140 @@
+//===----------------------------------------------------------------------===//
+//
+//                         Peloton
+//
+// settings_catalog.cpp
+//
+// Identification: src/catalog/settings_catalog.cpp
+//
+// Copyright (c) 2015-2017, Carnegie Mellon University Database Group
+//
+//===----------------------------------------------------------------------===//
+
+
+#include "catalog/settings_catalog.h"
+#include "catalog/catalog.h"
+#include "executor/logical_tile.h"
+#include "storage/data_table.h"
+#include "storage/tuple.h"
+
+#define SETTINGS_CATALOG_NAME "pg_settings"
+
+namespace peloton {
+namespace catalog {
+
+SettingsCatalog *SettingsCatalog::GetInstance(concurrency::Transaction *txn) {
+  static std::unique_ptr<SettingsCatalog> config_catalog(new SettingsCatalog(txn));
+  return config_catalog.get();
+}
+
+SettingsCatalog::SettingsCatalog(concurrency::Transaction *txn)
+        : AbstractCatalog("CREATE TABLE " CATALOG_DATABASE_NAME
+                          "." SETTINGS_CATALOG_NAME
+                          " ("
+                          "name   VARCHAR NOT NULL, "
+                          "value  VARCHAR NOT NULL, "
+                          "value_type   VARCHAR NOT NULL, "
+                          "description  VARCHAR, "
+                          "min_value    VARCHAR, "
+                          "max_value    VARCHAR, "
+                          "default_value    VARCHAR NOT NULL, "
+                          "is_mutable   BOOL NOT NULL, "
+                          "is_persistent  BOOL NOT NULL);",
+                          txn) {
+  // Add secondary index here if necessary
+  Catalog::GetInstance()->CreateIndex(
+          CATALOG_DATABASE_NAME, SETTINGS_CATALOG_NAME,
+          {"name"}, SETTINGS_CATALOG_NAME "_skey0",
+          false, IndexType::BWTREE, txn);
+}
+
+SettingsCatalog::~SettingsCatalog() {}
+
+bool SettingsCatalog::InsertSetting(
+    const std::string &name, const std::string &value,
```

Comment:
Yes, wrapping them into a struct is better but other catalogs do insert in this way so I don't modify it.